src/vy_adapter.cpp

```diff
@@ -0,0 +1,262 @@
+/* Copyright 2024 VyOS maintainers and contributors <maintainers@vyos.io>
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstring>
+#include <cstdlib>
+#include <vector>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <cstdint>
+#include <cassert>
+#include <algorithm>
+#include <list>
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <cstore/cstore.hpp>
+#include <vy_adapter.h>
+
+using namespace cstore;
+
+enum PIPES { READ, WRITE };
+
+extern FILE *out_stream;
+
+class stdout_redirect {
+    public:
+        stdout_redirect(): old_stdout(0), redirecting(false) {
+            out_pipe[READ] = 0;
+            out_pipe[WRITE] = 0;
+            if (pipe(out_pipe) == -1) {
+                return;
+            }
+
+            old_stdout = dup(fileno(stdout));
+            setbuf(stdout, NULL);
+            dup2(out_pipe[WRITE], fileno(stdout));
+
+            redirecting = true;
+        }
+
+        std::string get_redirected_output() {
+            if (!redirecting) {
+                return "";
+            }
+
+            char buffer[1024];
```

Comment:
This value can be arbitrary, but I wonder if a larger size can provide slightly better performance.